### DISCUSSION TITLE
Theme does not get applied to gtk Applications

### DISCUSSION THREAD

**Discussion by Dr0med4r** [nix-community/stylix#443] [Q&A]
I have a problem where neither the qt theme nor the gtk theme gets applied.
I use hyprland.
What i have done already:
- set GTK_THEME to adw-gtk3
- add adw-gtk3 to system packages
- set gtk-theme via dconf.settings in home-manager
here is my config https://gitlab.com/dr0med4r/nix-flake and the (I think important stylix config)
```nix
{
  lib,
  config,
  pkgs,
  username,
  ...
}: let
  cfg = config.home-modules.stylix;
in {
  options.home-modules.stylix = {
    enable = lib.mkEnableOption "enable stylix";
  };

  config = lib.mkIf cfg.enable {
    stylix = {
      enable = true;
      image = builtins.fetchurl {
        url = "https://w.wallhaven.cc/full/ex/wallhaven-ex9gwo.png";
        sha256 = "1xaqgn8n56q77ji22sf5134vqxbhgb8g6bw7rhr55bclnmzjdgph";
      };
      polarity = "dark";
      base16Scheme = "${pkgs.base16-schemes}/share/themes/tokyo-night-dark.yaml";
      # base16Scheme = "${pkgs.base16-schemes}/share/themes/nord.yaml";
      cursor = {
        package = pkgs.kdePackages.breeze;
        name = "breeze_cursors";
        size = 24;
      };
      opacity = {
        terminal = 0.8;
        desktop = 0.2;
        popups = 0.9;
      };
    };

    home-manager.users.${username} = {
      stylix = {
        targets = {
          kde.enable = true;
        };
      };
    };
  };
}


```
Did I do something wrong or is this a bug in stylix I don't have any ideas where to look anymore.

Ps: not setting gtk and dconf.settings changes nothing

**Comment by trueNAHO** [comment]
The following `gtk` directories are being generated [by your configuration](https://gitlab.com/Dr0med4r/nix-flake/-/tree/ffaf0340f344aa5e49c53bda7d65456f5fdd9734):

```console
$ nixos-rebuild build --flake .#test
$ fd --follow --type directory gtk result
result/etc/profiles/per-user/jakob/etc/gtk-2.0/
result/etc/profiles/per-user/jakob/etc/gtk-3.0/
result/etc/profiles/per-user/jakob/lib/gtk-2.0/
result/etc/profiles/per-user/jakob/lib/gtk-3.0/
result/etc/profiles/per-user/jakob/lib/gtk-4.0/
result/etc/profiles/per-user/jakob/share/gtksourceview-3.0/
result/etc/profiles/per-user/jakob/share/gtksourceview-4.0/
result/sw/etc/gtk-2.0/
result/sw/etc/gtk-3.0/
result/sw/lib/gtk-2.0/
result/sw/lib/gtk-3.0/
result/sw/lib/gtk-4.0/
result/sw/share/gtksourceview-3.0/
result/sw/share/gtksourceview-4.0/
result/sw/share/themes/Adwaita/gtk-2.0/
result/sw/share/themes/Adwaita/gtk-3.0/
result/sw/share/themes/Adwaita-dark/gtk-2.0/
result/sw/share/themes/Adwaita-dark/gtk-3.0/
result/sw/share/themes/Default/gtk-3.0/
result/sw/share/themes/Emacs/gtk-3.0/
result/sw/share/themes/Gtk/
result/sw/share/themes/HighContrast/gtk-2.0/
result/sw/share/themes/HighContrast/gtk-3.0/
```

Potentially related:

- https://github.com/danth/stylix/issues/400#issuecomment-2171592387
- https://github.com/danth/stylix/issues/445

**Reply by Dr0med4r** [reply]
You built the test configuration. This is an old config for testing DEs so it is not maintained and has stylix not enabled.
But I reran the commands with [this config](https://gitlab.com/Dr0med4r/nix-flake/-/tree/a22ad21a83bf2c3e957ed72c33791ae24259ccf4)
```
$ nixos-rebuild build --flake .#desktop
$ fd --follow --type directory gtk result
result/etc/profiles/per-user/jakob/etc/gtk-3.0/
result/etc/profiles/per-user/jakob/etc/gtk-2.0/
result/sw/etc/gtk-3.0/
result/sw/etc/gtk-2.0/
result/etc/profiles/per-user/jakob/lib/gtk-3.0/
result/etc/profiles/per-user/jakob/share/themes/adw-gtk3/
result/etc/profiles/per-user/jakob/lib/gtk-2.0/
result/etc/profiles/per-user/jakob/lib/gtk-4.0/
result/etc/profiles/per-user/jakob/share/themes/adw-gtk3/gtk-3.0/
result/etc/profiles/per-user/jakob/share/themes/adw-gtk3/gtk-4.0/
result/etc/profiles/per-user/jakob/share/themes/adw-gtk3-dark/
result/etc/profiles/per-user/jakob/share/themes/adw-gtk3-dark/gtk-3.0/
result/etc/profiles/per-user/jakob/share/themes/adw-gtk3-dark/gtk-4.0/
result/etc/profiles/per-user/root/etc/gtk-3.0/
result/etc/profiles/per-user/root/etc/gtk-2.0/
result/etc/profiles/per-user/root/share/themes/adw-gtk3/
result/etc/profiles/per-user/root/share/themes/adw-gtk3/gtk-3.0/
result/etc/profiles/per-user/root/share/themes/adw-gtk3/gtk-4.0/
result/etc/profiles/per-user/root/share/themes/adw-gtk3-dark/
result/etc/profiles/per-user/root/share/themes/adw-gtk3-dark/gtk-3.0/
result/etc/profiles/per-user/root/share/themes/adw-gtk3-dark/gtk-4.0/
result/etc/profiles/per-user/root/lib/gtk-3.0/
result/etc/profiles/per-user/root/lib/gtk-2.0/
result/etc/profiles/per-user/root/lib/gtk-4.0/
result/sw/lib/gtk-3.0/
result/sw/lib/gtk-2.0/
result/sw/lib/gtk-4.0/
```

And I would have expected a folder to be at `result/sw/share/themes/adw-gtk3/` and there is none.
But I am not sure where nix looks for themes so maybe these are not necessary.
Also all other programs are styled fine (btop, waybar, Alacritty, fish...)

**Comment by danth** [comment]
Is Home Manager creating `.config/gtk-4.0/gtk.css` in your home directory?

**Reply by Dr0med4r** [reply]
Yes I have a file from homemanager there and by looking with nix-inspect at `nixosConfigurations.tardus.config.home-manager.user.jakob.home.file."home/jakob/.config/gtk-4.0/gtk.css".source.buildCommand` I see the expected value from stylix.
And the file has the right contents.

**Comment by Dr0med4r** [comment]
I no have a partial answer.
It seems like the problem is the adw-gtk3 package.
If I switch to [this commit](https://gitlab.com/Dr0med4r/nix-flake/-/commit/707e51afc29a7b0687051151da3b6e32b21d29e0) and run `nix shell nixpkgs#awf -c awf-gtk3` via fuzzel it is the standard Adwaita theme.
But if I switch to [this commit](https://gitlab.com/Dr0med4r/nix-flake/-/commit/a37c8e1146e3a484c4b6220018bb23f090e0450c) and run the same command in fuzzel it is the correct theme.
If I try both in the terminal neither is themed correct.
The only thing changed is the adw-gtk3 package in `environment.systemPackages`

So it seems like some Variables are in fuzzel, which is run by Hyprland, that are not in the terminal.
@danth do you think this is a bug or more like a misconfiguration on my side. So should I open an issue or not.

**Comment by sonjiku** [comment]
Bumping this.
